manifests/plugin/chef.pp

```diff
@@ -0,0 +1,41 @@
+# = Foreman Proxy Chef plugin
+#
+# This class installs chef plugin
+#
+# === Parameters:
+#
+# $group::        group owner of the configuration file
+#
+# $enabled::      enables/disables the plugin
+#
+# $server_url::   chef server url
+#
+# $client_name::  chef client name used for authentication of other client requests
+#
+# $private_key::  path to file containing private key for $client_name client
+#
+# $ssl_verify::   should we perform chef server ssl cert verification? this requires
+#                 CA certificate installed and trusted
+#                 type:boolean
+#
+# $ssl_pem_file:: if $ssl_verify is true you can specify a path to a file which
+#                 contains certificate and related private key if the certificate
+#                 is not globally trusted
+#
+class foreman_proxy::plugin::chef (
+  $enabled      = $::foreman_proxy::plugin::chef::params::enabled,
+  $group        = $::foreman_proxy::plugin::chef::params::group,
+  $server_url   = $::foreman_proxy::plugin::chef::params::server_url,
+  $client_name  = $::foreman_proxy::plugin::chef::params::client_name,
+  $private_key  = $::foreman_proxy::plugin::chef::params::private_key,
+  $ssl_verify   = $::foreman_proxy::plugin::chef::params::ssl_verify,
+  $ssl_pem_file = $::foreman_proxy::plugin::chef::params::ssl_pem_file,
+) inherits foreman_proxy::plugin::chef::params {
+  $group_real = pick($group, $::foreman_proxy::user)
```

Comment:
You can leave this out (and the validate_string) since the settings file defaults to the correct value and you already have undef in params.pp.